### PR TITLE
Rc1.1.8

### PR DESCRIPTION
**Pull request by coderidge**
Includes new copy and 2 new Drinking specifications 2 - 4 days <= 14 units , 2- 4 days 15+ units men and women

### CODE REVIEW

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php [hunk 1]
```diff
@@ -51,6 +53,35 @@ public function buildNarrative(Questionnaire $questionnaire)
         return $this->narrative;
     }
 
+             /**
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php]: Formatting not psr2 please fix

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php [hunk 2]
```diff
@@ -63,24 +94,29 @@ protected function buildGreenDrinkingNarrative(Questionnaire $questionnaire)
             case DrinkingIsLTMonthlyAndHasBingeSpecification::isSatisfiedBy($questionnaire):
                 $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_lt_monthly_with_binge'));
                 break;
+            case Drinking2to4TimesAMonthDoesNotExceed14UnitsAWeekSpecification::isSatisfiedBy($questionnaire):
+                $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_2_to_4_monthly_14_or_less_units'));
+                break;
             case DrinkingLessThanMonthlyAndDoesNotBingeSpecification::isSatisfiedBy($questionnaire):
-                $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_lt_monthly_no_binge'));
+               $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_lt_monthly_no_binge'));
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php]: Indentation is out

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php [hunk 3]
```diff
@@ -63,24 +94,29 @@ protected function buildGreenDrinkingNarrative(Questionnaire $questionnaire)
...
                 break;
             case Drinking2to4TimesAMonthWithBingeSpecification::isSatisfiedBy($questionnaire):
                 $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_2_to_4_monthly_with_binge'));
                 break;
             case Drinking2to4TimesAMonthNoBingeSpecification::isSatisfiedBy($questionnaire):
                 $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_2_to_4_monthly_no_binge'));
                 break;
-            case DrinkingLTE2DaysLTE3UnitsSpecification::isSatisfiedBy($questionnaire):
+             case DrinkingLTE2DaysLTE3UnitsSpecification::isSatisfiedBy($questionnaire):
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php]: as above

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php [hunk 4]
```diff
@@ -63,24 +94,29 @@ protected function buildGreenDrinkingNarrative(Questionnaire $questionnaire)
...
                 $this->narrative->addMessages(self::SECTION_NAME, array('drinking_green_lt_2_days'));
                 break;
-            default:
+             default:
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/Drinking/DrinkingNarrativeBuilder.php]: as above

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/Drinking2to4TimesAMonthDoesExceed15UnitsAWeekForGenderSpecification.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+{
+
+    public static function isSatisfiedBy(Questionnaire $questionnaire)
+    {
+
+        $drinking = $questionnaire->getDrinking();
+
+        $gender = $questionnaire->getPerson()->getGender();
+
+        switch ($gender) {
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/Drinking2to4TimesAMonthDoesExceed15UnitsAWeekForGenderSpecification.php]: You could, for future reference write this as `switch ($questionnaire->getPerson()->getGender())

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/Drinking2to4TimesAMonthDoesNotExceed14UnitsAWeekSpecification.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    
+    public static function isSatisfiedBy(Questionnaire $questionnaire)
+    {
+       $drinking = $questionnaire->getDrinking();
+        
+        return (
+             DrinkingScore::RAG_GREEN_CUTOFF >= DrinkingScore::getWeeklyUnits($drinking)
+             &&
+             DrinkingMonthlyOrLessSpecification::isSatisfiedBy($questionnaire)
+             );
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/Drinking2to4TimesAMonthDoesNotExceed14UnitsAWeekSpecification.php]: align closing parenthesis with return

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/DrinkingTwoToFourTimesAMonthSpecification.php [hunk 7]
```diff
@@ -17,7 +17,7 @@ class DrinkingTwoToFourTimesAMonthSpecification implements NarrativeSpecificatio
      */
     public static function isSatisfiedBy(Questionnaire $questionnaire)
     {
-        return $questionnaire->getDrinking()->getDoYouDrink() === Drinking::DRINK_TWO_TO_FOUR_TIMES_A_MONTH;
+    return $questionnaire->getDrinking()->getDoYouDrink() === Drinking::DRINK_TWO_TO_FOUR_TIMES_A_MONTH;
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Specification/Drinking/DrinkingTwoToFourTimesAMonthSpecification.php]: indentation

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Scoring/Rag/DrinkingScore.php [hunk 8]
```diff
@@ -224,19 +224,18 @@ public static function getDailyMaxUnits(Drinking $drinking)
      */
     protected function generateScore(Questionnaire $questionnaire)
     {
-
         switch (true) {
             case true === $this->ragIsGreen($questionnaire):
-                $rag   = self::RAG_GREEN;
-                $score = self::SCORE_GREEN;
-                break;
+            $rag   = self::RAG_GREEN;
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Scoring/Rag/DrinkingScore.php]: indent the sub statements under the conditions

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Scoring/Rag/DrinkingScore.php [hunk 9]
```diff
@@ -249,6 +248,8 @@ protected function generateScore(Questionnaire $questionnaire)
     protected function ragIsGreen(Questionnaire $questionnaire)
     {
         switch (true) {
+            //case $this->isDoTheyDrinkLessThan14units($questionnaire):
```
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Scoring/Rag/DrinkingScore.php]: remove the commented out lines, also, if you are keeping the functions, rename them to isGreenAs<condition>

**Comment by brettminnie** [comment]
You can just add to this pull request, update and push, will merge once formatting is fixed

**Comment by coderidge** [comment]
Ok thanks, believe I've changed formatting

**Comment by coderidge** [comment]
Hi, PHE were talking about 10am deadline to review changes so would be good to merge to staging as soon as possible, thanks

**Comment by brettminnie** [comment]
I'll get it into staging and send you a whitespace patch for a new pr

**Pull request merged**